users.cpp

```diff
@@ -194,5 +198,79 @@ bool Users::userPasswordExpired(void) const
     return manager.userPasswordExpired(userName);
 }
 
+void Users::createSecretKey()
+{
+    if (!std::filesystem::exists(authAppPath))
+    {
+        lg2::error("No authenticator app found at {PATH}", "PATH", authAppPath);
+        return;
+    }
+    std::string path = std::format(secretKeyTempPath, userName);
+    executeCmd(authAppPath.data(), "-s", path.c_str(), "-u", "-W", "-Q", "NONE",
```

Comment:
This created temp file. But where is the permanent secret key file is created ? There should be a rename of this file to the permanent file right ?